<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DlgAtlasGroupSettings</class>
 <widget class="QDialog" name="DlgAtlasGroupSettings">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>296</width>
    <height>436</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Atlas Bank Settings</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Bank Name:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="txtName"/>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Packing Options</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QFormLayout" name="formLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>Sort images by:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="cmbSortOrder">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Sorting of images significantly increases packing effectivity.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="currentIndex">
           <number>0</number>
          </property>
          <item>
           <property name="text">
            <string>Largest edges</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Total area</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Widths</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Heights</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Don't sort</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Heuristic:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QComboBox" name="cmbHeuristic">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Weight function tries to find best place for each image. Top left will be best in most cases.&lt;/p&gt;&lt;p&gt;Some heuristics depend on sorting method.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="currentIndex">
           <number>1</number>
          </property>
          <item>
           <property name="text">
            <string>None</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Top left</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Best area</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Best short side</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Best long side</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Min width</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Min height</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QGridLayout" name="gridLayout">
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <property name="spacing">
         <number>0</number>
        </property>
        <item row="2" column="2">
         <widget class="QSpinBox" name="sbFrameMarginTop">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="maximumSize">
           <size>
            <width>60</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="readOnly">
           <bool>false</bool>
          </property>
          <property name="accelerated">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="4" column="2">
         <widget class="QSpinBox" name="sbFrameMarginBottom">
          <property name="maximumSize">
           <size>
            <width>60</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="value">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item row="3" column="3">
         <widget class="QSpinBox" name="sbFrameMarginRight">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>60</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="value">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QSpinBox" name="sbFrameMarginLeft">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>60</width>
            <height>16777215</height>
           </size>
          </property>
         </widget>
        </item>
        <item row="3" column="2">
         <widget class="QLabel" name="label_13">
          <property name="toolTip">
           <string>Border around each image. If atlas will be saled (OpenGL), keep at least 1 px border.</string>
          </property>
          <property name="text">
           <string>Margin</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QCheckBox" name="chkSquareTexturesOnly">
        <property name="text">
         <string>Square Textures Only</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QLabel" name="label_3">
          <property name="toolTip">
           <string>Maximum size of resulting packed texture in pixels.</string>
          </property>
          <property name="text">
           <string>Max Texture Size: </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="sbTextureWidth">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimum">
           <number>32</number>
          </property>
          <property name="maximum">
           <number>16384</number>
          </property>
          <property name="singleStep">
           <number>2</number>
          </property>
          <property name="value">
           <number>4096</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnMatchTextureWidthHeight">
          <property name="maximumSize">
           <size>
            <width>20</width>
            <height>20</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Swap width and height</string>
          </property>
          <property name="text">
           <string>x</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="sbTextureHeight">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimum">
           <number>32</number>
          </property>
          <property name="maximum">
           <number>16384</number>
          </property>
          <property name="singleStep">
           <number>2</number>
          </property>
          <property name="value">
           <number>4096</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QPushButton" name="btnTexSize128">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>46</width>
            <height>20</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>8</pointsize>
           </font>
          </property>
          <property name="text">
           <string>128</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnTexSize256">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>46</width>
            <height>20</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>8</pointsize>
           </font>
          </property>
          <property name="text">
           <string>256</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnTexSize512">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>46</width>
            <height>20</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>8</pointsize>
           </font>
          </property>
          <property name="text">
           <string>512</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnTexSize1024">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>46</width>
            <height>20</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>8</pointsize>
           </font>
          </property>
          <property name="text">
           <string>1024</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QPushButton" name="btnTexSize2048">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>46</width>
            <height>20</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>8</pointsize>
           </font>
          </property>
          <property name="text">
           <string>2048</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnTexSize4096">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>46</width>
            <height>20</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>8</pointsize>
           </font>
          </property>
          <property name="text">
           <string>4096</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnTexSize8192">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>46</width>
            <height>20</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>8</pointsize>
           </font>
          </property>
          <property name="text">
           <string>8192</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnTexSize16384">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>46</width>
            <height>20</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>8</pointsize>
           </font>
          </property>
          <property name="text">
           <string>16384</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QGroupBox" name="grpCropUnusedSpace">
        <property name="title">
         <string>Crop Unused Space</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QCheckBox" name="chkAggressiveResize">
           <property name="toolTip">
            <string>May create more textures overall, but might waste less space</string>
           </property>
           <property name="text">
            <string>Aggressive resizing (more smaller textures)</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLabel" name="lblResizingThreshold">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Minimum Fill Rate</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="sbResizingThreshold">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="suffix">
              <string>%</string>
             </property>
             <property name="minimum">
              <number>50</number>
             </property>
             <property name="maximum">
              <number>95</number>
             </property>
             <property name="value">
              <number>80</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="chkForceRegenBank">
     <property name="text">
      <string>Force Regenerate Bank</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>DlgAtlasGroupSettings</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>DlgAtlasGroupSettings</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
